src/Lecture1.hs

```diff
@@ -98,7 +105,11 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [Char] -> [Char]
+subString start end str
+    | end < 0 || start > end = []
+    | start < 0 = subString 0 end str
```

Comment:
You can slightly simplify this by using `take` directly and avoiding a few extra calls 🙂 
```suggestion
    | start < 0 = take (end + 1) str
```